### ISSUE TITLE
[firebase_analytics: 11.3.0]: In Flutter version 3.24.2: Error (Xcode): Linker command failed with exit code 1

### ISSUE THREAD

**Issue by devnta** [firebase/flutterfire#13277]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Description

When I update to firebase_analytics 11.3.0 getting an error.
```
Error (Xcode): Undefined symbols:

Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)

Could not build the application for the simulator.
Error launching application on iPhone 15 Pro Max.
```
![image](https://github.com/user-attachments/assets/41597570-8185-4129-941f-baa21ef52a3f)

### Reproducing the issue

1. Create new Flutter project.
2. Add to `pubspec.yaml`:
```
  firebase_core: ^3.4.0
  firebase_analytics: ^11.3.0
```
3. Update `ios/Podfile` to:
```
platform :ios, '13.0'
```
4. Run command:
- flutter pub get
- cd to /ios and run `pod install`.
5. Run project in iOS simulator with Flutter version 3.24.2.

### Firebase Core version

3.4.0

### Flutter Version

3.24.2

### Flutter dependencies

<details>
<summary>Expand <code>Flutter dependencies</code> snippet</summary>
<br>

```yaml

Replace this line with the contents of your `flutter pub deps -- --style=compact`.

```

</details>

**Comment by SelaseKay** [comment]
Hi @phamquoctrongnta , this issue does not seem to be pointing to `firebase_analytics` in any way. I'm going to go ahead and close this issue since it's out of scope for this repo.

**Comment by phamquoctrongnta** [comment]
@SelaseKay 
When I delete `firebase_analytics`, my project works. Does this related?

**Comment by SelaseKay** [comment]
Try deleting your `Podfile.lock` and running `pod install`.

**Comment by phamquoctrongnta** [comment]
@SelaseKay 
I ran `pod install` with no problem. But when I ran the project, I got an error.
Which platform did you check? Android or iOS? Please check on an iOS device.

**Comment by phamquoctrongnta** [comment]
@SelaseKay 
Reproducing the issue:
1. Create new Flutter project.
2. Add to `pubspec.yaml`:
```
  firebase_core: ^3.4.0
  firebase_analytics: ^11.3.0
```
3. Update `ios/Podfile` to:
```
platform :ios, '13.0'
```
4. Run command:
- flutter pub get
- cd to /ios and run `pod install`.
5. Run project in iOS simulator with Flutter version 3.24.2.
 As you can see, I didn’t change anything, but the project still getting an error when I run it.

**Comment by RoberRaf** [comment]
I have a quit similar issue.
the error message is : 
Error (Xcode): Undefined symbol: _OBJC_CLASS_$_FIRMessaging

Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)

this issue appeared only after updating to flutter 3.24.2

**Comment by SelaseKay** [comment]
What Xcode version are you experiencing this on?

**Comment by hasen6** [comment]
> What Xcode version are you experiencing this on?

15.0.1

It's the same for me as the OP, removing firebase_analytics from dependencies enables the app to load fine, it's definitely due to firebase_analytics. Also the error on an xcode build for me is:

fatal error: module 'firebase_analytics' not found
@import firebase_analytics;

**Comment by SelaseKay** [comment]
I'm unable to reproduce this on Xcode version 15.2. Kindly try updating your Xcode and test again.

**Comment by hasen6** [comment]
> I'm unable to reproduce this on Xcode version 15.2. Kindly try updating your Xcode and test again.

Yes that fixed the problem. Please update your documentation to make it clear xcode 15.2 or more is required (if it's not really there). Not sure what version the OP was using but it's likely it would solve the problem too.

**Comment by RoberRaf** [comment]
> I'm unable to reproduce this on Xcode version 15.2. Kindly try updating your Xcode and test again.

for me I' using xcode version 15.4
the last stable version 

is there in keys ?

**Comment by SelaseKay** [comment]
@RoberRaf , does version 15.4 resolve your issue?

**Comment by RoberRaf** [comment]
> @RoberRaf , does version 15.4 resolve your issue?

I resolved the problem by re-using an  ios folder of an old version of  my project.
It wasn't related to xcode version, nor firebase plugins. I think It was related to some xcode configuration. Can't specify which one.

Thank you anyway 🌸

**Comment by youpelegrace** [comment]
Hello is anyone still facing this issue? 

i'm having this error and i have tried xcode 15.2 and 15.3 nd also 15 but the error persists

 Could not find or use auto-linked library 'swiftXPC': library 'swiftXPC' not found
ld: warning: Could not find or use auto-linked framework 'CoreAudioTypes': framework 'CoreAudioTypes' not found
ld: Undefined symbols:
  _FIRConsentStatusDenied, referenced from:
      -[FLTFirebaseAnalyticsPlugin setConsent:withMethodCallResult:] in firebase_analytics[4](FLTFirebaseAnalyticsPlugin.o)
  _FIRConsentStatusGranted, referenced from:
      -[FLTFirebaseAnalyticsPlugin setConsent:withMethodCallResult:] in firebase_analytics[4](FLTFirebaseAnalyticsPlugin.o)
  _FIRConsentTypeAdPersonalization, referenced from:
      -[FLTFirebaseAnalyticsPlugin setConsent:withMethodCallResult:] in firebase_analytics[4](FLTFirebaseAnalyticsPlugin.o)
  _FIRConsentTypeAdStorage, referenced from:
      -[FLTFirebaseAnalyticsPlugin setConsent:withMethodCallResult:] in firebase_analytics[4](FLTFirebaseAnalyticsPlugin.o)
  _FIRConsentTypeAdUserData, referenced from:
      -[FLTFirebaseAnalyticsPlugin setConsent:withMethodCallResult:] in firebase_analytics[4](FLTFirebaseAnalyticsPlugin.o)
  _FIRConsentTypeAnalyticsStorage, referenced from:
      -[FLTFirebaseAnalyticsPlugin setConsent:withMethodCallResult:] in firebase_analytics[4](FLTFirebaseAnalyticsPlugin.o)
  _OBJC_CLASS_$_FIRAnalytics, referenced from:
       in firebase_analytics[4](FLTFirebaseAnalyticsPlugin.o)
  __swift_FORCE_LOAD_$_swiftXPC, referenced from:
      __swift_FORCE_LOAD_$_swiftXPC_$_FBAudienceNetwork in FBAudienceNetwork[415](FBAudienceNetwork.o)
clang: error: linker command failed with exit code 1 (use -v to see invocation)

this is my error

**Comment by hafidzaini** [comment]
same error with me.
CoreAudioType and undefined symbol

**Comment by youpelegrace** [comment]
@hafidzaini you might want to remove firebase analytics plugin and run the app, if it runs successfully 
add $FirebaseSDKVersion = '10.29.0' to your podfile and re add the plugin back

**Comment by hafidzaini** [comment]
thanks, i'try.

this problem happen after i do flutter clean and then do arch -x86_64 pod install on mac M1
last week still no problem and didnt change any plugin version and didnt update xcode.
but it was before a warning on xcode "Conversion to Swift available"
now its break. strange.. 😰🤔

**Comment by hafidzaini** [comment]
never change package version, this error happen after flutter clean. 
Solve this issue by update xCode to latest version, from 15.0 to 15.4

**Comment by afrish** [comment]
Why is this closed? I have this problem with the newest Flutter 3.24.3 and the newest Firbase versions from pub.dev, it's reproduced 100% of the time, i.e. the build always fails

**Comment by iamriajul** [comment]
Same, in a new Flutter project using 3.24.3.
When adding firebase_analytics the projects fails to build, but not when using firebase_messaging, firebase_crashlytics.

**Comment by afrish** [comment]
> Same, in a new Flutter project using 3.24.3. When adding firebase_analytics the projects fails to build, but not when using firebase_messaging, firebase_crashlytics.

The problem was solved for me when I updated Xcode from 15.0 to 16.0. There are several questions though:
1. Why was it failing with the previous version?
2. Why was it not mentioned anywhere in the error message that Xcode needs to be updated?
3. Why the Xcode update was not even available until I updated the whole OS?

**Comment by GhassanAlKaraan** [comment]
I'm encountering the same issue with Flutter 3.24.3 on Xcode 15.0. I've tried all possible solutions with the Podfile and iOS/Xcode configurations.

The error only occurs when adding firebase_analytics 11.3.3.

It seems like upgrading to Xcode 16.0 (the latest version) is the only solution. However, I can't upgrade right now because I have projects with older Flutter versions that aren't compatible with Xcode 16. For now, I'll likely have to skip using firebase_analytics.

**Comment by devnta** [comment]
@GhassanAlKaraan 
Based on the reply in https://github.com/firebase/flutterfire/issues/13277#issuecomment-2331048836, I understand that this issue is related to Xcode, not to firebase_analytics. :D :D :D 
I think this issue needs to be re-opened because it is directly related to `firebase_analytics`
